zips/zip-0312.rst

```diff
@@ -141,11 +141,56 @@ concretely in the Ciphersuites section.
-it needs to be consistent with FROST, see [#frost-tdkg]_ for guidance. The spend
-authorization private key :math:`\mathsf{ask}` [#protocol-spendauthsig]_ is the
-particular key that must be used in the context of this ZIP. Note that the
-:math:`\mathsf{ask}` is usually derived from the spending key
+While FROST key generation is out of scope for this ZIP and the FROST spec
+[#FROST]_, it needs to be consistent with FROST; see [#frost-tdkg]_ for general
+guidance. This section assumes all participants have run either a trusted dealer
+or distributed key generation process, and each participant has their signing
+key share `sk_i` and the group public key `PK`. Trusted dealer key generation
+MAY be used; distributed key generation SHOULD be used.
+
+To define a spending or viewing key that uses FROST, the Sapling and Orchard key
+trees [#protocol-saplingkeycomponents]_ [#protocol-orchardkeycomponents]_ are
+adjusted as follows:
+
+- The Spend validating key :math:`\mathsf{ak}` is replaced by the FROST group
+  public key `PK` [#frost-protocol]_.
+- The Spend authorizing key :math:`\mathsf{ask}` is replaced by the logical
+  signing key that corresponds to the group public key `PK`. By design, this key
+  is never explicitly constructed (unless trusted dealer key generation was
+  used), and instead is represented by any sufficient subset of the participant
+  FROST signing key shares `sk_i`.
+
+The remaining parts of the Sapling and Orchard key trees SHOULD be generated
+from a common spending key :math:`\mathsf{sk}` as described in
+[#protocol-saplingkeycomponents]_ [#protocol-orchardkeycomponents]_, with the
+exception of :math:`\mathsf{ask}` which SHOULD NOT be derived from
+:math:`\mathsf{sk}` (and can't possibly be derived, if distributed key
+generation was used). Deriving :math:`\mathsf{ask}` from :math:`\mathsf{sk}` MAY
+be done if the application requires to backup the unsplit key, but this is
+generally not recommended since it forces the usage of trusted dealer key
+generation.
+
+In order for all participants to agree on the value of :math:`\mathsf{sk}`, one
+of the following options SHOULD be carried out:
+
+- One participant generates a random :math:`\mathsf{sk}` and sends it to the
+  others via an encrypted and authenticated channel. This option SHOULD be used
+  if trusted dealer key generation was used, and the :math:`\mathsf{sk}` value
+  can be sent along with the FROST shares during key generation. This option MAY
+  be used if distributed key generation was used, if participants find it
+  acceptable to trust one participant to correctly generate :math:`\mathsf{sk}`.
+- Participants generate a :math:`\mathsf{sk}` in distributed manner using TODO.
+  This option SHOULD be used if distributed key generation was used.
+- (TODO: evaluate if it is secure) Participants derive :math:`\mathsf{sk}` from
+  the public keys of the particpants (`PK_i` in [#FROST]_) via (TODO: specify
+  mechanism to hash all PK_i and derive sk)
```

Comment:
In the R&D Meeting today, @daira, @ebfull, @nuttycom and myself were discussing post-quantum Zcash preparations, and part of the approach we are discussing requires that `ivk` can be treated as a commitment to `sk`. I raised the point that with this draft proposal for FROST keygen, this is not the case. After discussion, we decided to suggest here that for FROST viewing keys, `rivk` be derived from `(sk, ak, nk)` instead of just `sk`. This is sufficient to ensure that `Commit^ivk` is effectively post-quantum binding (the argument is complicated, but a draft ZIP is forthcoming).

Reply:
I don't see any problems. Is there a concrete suggestion on how to do the derivation?